src/Service/QueueManager.php

```diff
@@ -346,4 +347,32 @@ public function isSuccessfulStatusCode(?int $statusCode): bool
     {
         return $statusCode !== null && $statusCode >= 200 && $statusCode < 300;
     }
+
+    /**
+     * @param string $batchId
+     * @return array<string, JobInterface>
+     */
+    private function getBatch(string $batchId): array
+    {
+        return $this->batches[$batchId] ?? [];
+    }
+
+    private function addJobToBatch(string $batchId, JobInterface $job): void
+    {
+        $this->batches[$batchId][$job->getUuid()] = $job;
+    }
+
+    private function removeJobFromBatch(string $batchId, string $jobUuid): void
+    {
+        unset($this->batches[$batchId][$jobUuid]);
+    }
+
+    private function emptyBatch(string $batchId): void
```

Comment:
```suggestion
    private function clearBatch(string $batchId): void
```
I would rename this method to "clearBatch" to make it clearer that it does not test whether a batch is empty, it actually clears it.